### PR TITLE
TMA-1025: Parametrize output stage prefix

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: spec/unit/models/project_spec.rb [hunk 1]
```diff
@@ -49,6 +61,80 @@
...
       end
     end
+
+    it 'creates output stage with the same prefix as in the original project' do
+      expect(GoodData::AdsOutputStage).to receive(:create)
+        .with(hash_including(output_stage_prefix: original_prefix))
+      subject.transfer_output_stage(from_project, to_project, {})
+    end
+
+    context 'when ads_output_stage_prefix is specified' do
```
Comment (yenkins) [inline spec/unit/models/project_spec.rb]: warning: Similar code found in project_spec.rb:71, project_spec.rb:83 (mass = 60) (flay)

File: spec/unit/models/project_spec.rb [hunk 2]
```diff
@@ -49,6 +61,80 @@
...
+      end
+
+      it 'creates output stage with the specified prefix' do
+        expect(GoodData::AdsOutputStage).to receive(:create)
+          .with(hash_including(output_stage_prefix: prefix))
+        subject.transfer_output_stage(from_project, to_project, options)
+      end
+    end
+
+    context 'when ads_output_stage_uri is specified' do
```
Comment (yenkins) [inline spec/unit/models/project_spec.rb]: warning: Similar code found in project_spec.rb:71, project_spec.rb:83 (mass = 60) (flay)

File: spec/unit/models/project_spec.rb [hunk 3]
```diff
@@ -49,6 +61,80 @@
...
+        allow(to_output_stage).to receive(:output_stage_prefix=)
+      end
+
+      it 'sets output stage prefix to the same value as in the original project' do
+        expect(to_output_stage).to receive(:output_stage_prefix=).with(original_prefix)
+        expect(to_output_stage).to receive(:save)
+        subject.transfer_output_stage(from_project, to_project, {})
+      end
+
+      context 'when ads_output_stage_prefix is specified' do
```
Comment (yenkins) [inline spec/unit/models/project_spec.rb]: warning: Similar code found in project_spec.rb:114, project_spec.rb:126 (mass = 62) (flay)

File: spec/unit/models/project_spec.rb [hunk 4]
```diff
@@ -49,6 +61,80 @@
...
+        end
+
+        it 'sets output stage prefix to the specified value' do
+          expect(to_output_stage).to receive(:output_stage_prefix=).with(prefix)
+          expect(to_output_stage).to receive(:save)
+          subject.transfer_output_stage(from_project, to_project, options)
+        end
+      end
+
+      context 'when ads_output_stage_uri is specified' do
```
Comment (yenkins) [inline spec/unit/models/project_spec.rb]: warning: Similar code found in project_spec.rb:114, project_spec.rb:126 (mass = 62) (flay)

File: lib/gooddata/models/project.rb [hunk 5]
```diff
@@ -195,13 +195,15 @@ def user_csv_import(row)
       end
 
       def transfer_output_stage(from_project, to_project, options)
+        from_prj_output_stage = from_project.add.output_stage
+        output_stage_prefix = options[:ads_output_stage_prefix] || from_prj_output_stage.output_stage_prefix
+        output_stage_uri = options[:ads_output_stage_uri] || from_prj_output_stage.schema
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: info: Calls 'options[:ads_output_stage_uri]' 2 times (DuplicateMethodCall) (reek)

File: lib/gooddata/models/project.rb [hunk 6]
```diff
@@ -195,13 +195,15 @@ def user_csv_import(row)
...
         if from_project.processes.any? { |p| p.type == :dataload }
           if to_project.processes.any? { |p| p.type == :dataload }
-            to_project.add.output_stage.schema = from_project.add.output_stage.schema
-            to_project.add.output_stage.output_stage_prefix = from_project.add.output_stage.output_stage_prefix
+            to_project.add.output_stage.schema = output_stage_uri
```
Comment (yenkins) [inline lib/gooddata/models/project.rb]: info: Calls 'to_project.add' 4 times (DuplicateMethodCall) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/593/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1414/ : FAILURE in 2m 51s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1684/ : FAILURE in 2m 46s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/595/ : SUCCESS in 41s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1417/ : FAILURE in 2m 41s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1687/ : FAILURE in 2m 41s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/596/ : SUCCESS in 32s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1418/ : FAILURE in 1m 11s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1688/ : FAILURE in 1m 12s

File: spec/double_with_class.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2018 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module RSpec
+  module Mocks
+    module ExampleMethods
```
Comment (yenkins) [inline spec/double_with_class.rb]: info: Has no descriptive comment (IrresponsibleModule) (reek)

File: spec/double_with_class.rb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+      # Creates a double and mocks its #class method to match the object
+      def double_with_class(klass)
+        d = double(klass)
```
Comment (yenkins) [inline spec/double_with_class.rb]: info: Has the variable name 'd' (UncommunicativeVariableName) (reek)

File: spec/unit/datawarehouse_mock.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2018 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module GoodData
+  class Datawarehouse
```
Comment (yenkins) [inline spec/unit/datawarehouse_mock.rb]: info: Has no descriptive comment (IrresponsibleModule) (reek)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/599/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1421/ : FAILURE in 2m 44s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1691/ : FAILURE in 1m 24s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/600/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1422/ : FAILURE in 1m 07s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1692/ : FAILURE in 2m 29s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/601/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1423/ : FAILURE in 1m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1693/ : FAILURE in 1m 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/602/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1424/ : FAILURE in 1m 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1694/ : FAILURE in 1m 15s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/603/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1425/ : FAILURE in 1m 06s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1695/ : FAILURE in 1m 13s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/605/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1427/ : SUCCESS in 1m 11s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1697/ : SUCCESS in 1m 12s

**Comment by kubamahnert** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/606/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1526/ : SUCCESS in 34m 16s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/95/ : SUCCESS in 26m 41s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/131/ : SUCCESS in 23s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1432/ : FAILURE in 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1701/ : SUCCESS in 1m 10s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1528/ : FAILURE in 2m 08s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-lcm-integration-tests-zuul/97/ : SUCCESS in 27m 38s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/697/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1524/ : FAILURE in 1m 35s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1785/ : FAILURE in 1m 39s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/24/ : SUCCESS in 6m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-opensource-repository-check/699/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/1526/ : FAILURE in 1m 12s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/1787/ : FAILURE in 1m 08s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-check-pipeline/26/ : SUCCESS in 5m 29s

**Comment by panjan** [comment]
https://travis-ci.org/panjan/gooddata-ruby/builds/437555122

**Pull request merged**